### PR TITLE
deprecate(checkbox, radio-button): deprecate 'guid' properties

### PR DESCRIPTION
**Pull request by driskull**
**Related Issue:** #9110

## Summary

- deprecate guid properties

### CODE REVIEW

**Comment by jcfranco** [comment]
Thanks to @DitwanP and @benelan, we already do! https://github.com/Esri/calcite-design-system/pull/10346

**Review by geospatialem — APPROVED** [review]
Awesome! 💥 

Did you want to add this into the September release today, or hold off until merging tomorrow for our next release (`3.0`)?

**Comment by driskull** [comment]
September would be good.

**Pull request merged**
